internal/engine/local_target_build_and_deployer.go

```diff
@@ -0,0 +1,74 @@
+package engine
+
+import (
+	"context"
+	"os/exec"
+
+	"github.com/windmilleng/tilt/internal/store"
+	"github.com/windmilleng/tilt/pkg/logger"
+	"github.com/windmilleng/tilt/pkg/model"
+)
+
+var _ BuildAndDeployer = &LocalTargetBuildAndDeployer{}
+
+// TODO(maia): CommandRunner interface for testability
+type LocalTargetBuildAndDeployer struct{}
+
+func NewLocalTargetBuildAndDeployer() *LocalTargetBuildAndDeployer {
+	return &LocalTargetBuildAndDeployer{}
+}
+
+func (bd *LocalTargetBuildAndDeployer) BuildAndDeploy(ctx context.Context, st store.RStore, specs []model.TargetSpec, stateSet store.BuildStateSet) (resultSet store.BuildResultSet, err error) {
+	targets := bd.extract(specs)
+	if len(targets) != 1 {
+		return store.BuildResultSet{}, SilentRedirectToNextBuilderf(
+			"LocalTargetBuildAndDeployer requires exactly one LocalTarget (got %d)", len(targets))
+	}
+
+	targ := targets[0]
+	err = bd.run(ctx, targ.Cmd)
+	if err != nil {
+		// (Never fall back from the LocalTargetBaD, none of our other BaDs can handle this target)
+		return store.BuildResultSet{}, DontFallBackErrorf("Command %q failed: %v", targ.Cmd.String(), err)
+	}
+
+	return bd.successfulBuildResult(targ), nil
+}
+
+// Extract the targets we can apply -- i.e. LocalTargets
+func (bd *LocalTargetBuildAndDeployer) extract(specs []model.TargetSpec) []model.LocalTarget {
+	var targs []model.LocalTarget
+	for _, s := range specs {
+		switch s := s.(type) {
+		case model.LocalTarget:
+			targs = append(targs, s)
+		}
+	}
+	return targs
+}
+
+func (bd *LocalTargetBuildAndDeployer) run(ctx context.Context, c model.Cmd) error {
+	if len(c.Argv) == 0 {
+		panic("LocalTargetBuildAndDeployer tried to run empty command " +
+			"(should have been caught by Validate() )")
+	}
+
+	writer := logger.Get(ctx).Writer(logger.InfoLvl)
+	cmd := exec.CommandContext(ctx, c.Argv[0], c.Argv[1:]...)
```

Comment:
Would be good to say what the command is before running it, and maybe a pipelinestate with timing info

Reply:
good call -- in the interest of getting this working E2E i'll leave a TODO/make a ticket

Reply:
I think for debugging at least printing the Argv would be helpful.